cmd/metal-api/internal/service/firmware-service.go

```diff
@@ -137,22 +141,26 @@ func (r *firmwareResource) uploadFirmware(request *restful.Request, response *re
 		Key:    &key,
 		Body:   file,
 	})
-	if checkError(request, response, utils.CurrentFuncName(), err) {
+	if err != nil {
+		r.sendError(request, response, httperrors.InternalServerError(err))
```

Comment:
here we have InternalServerError instead of defaultError or Badrequest, confusing

Reply:
Not required because not accessing DB or masterdata-api.